src/radix_connect/mobile/client.rs

```diff
@@ -1,65 +1,189 @@
+    /// Once the session is validated, it will be moved to the secure storage.
+    /// Validation consists in verifying the origin of the session.
+    new_sessions: RwLock<HashMap<SessionID, Session>>,
+}
+
+impl RadixConnectMobile {
+    pub fn init(
+        wallet_interactions_transport: Arc<dyn WalletInteractionTransport>,
+        session_storage: Arc<dyn SessionStorage>,
+    ) -> Self {
+        Self {
+            wallet_interactions_transport,
+            session_storage,
+            new_sessions: RwLock::new(HashMap::new()),
+        }
+    }
+}
 
-/// The Radix Connect Mobile client.
-/// This is the object that will be used by the mobile app to handle interactions sent over Radix Connect Relay.
-// #[derive(uniffi::Object)]
-// pub struct RadixConnectMobile {}
-
-// // Provisional API
-// #[uniffi::export]
-// impl RadixConnectMobile {
-//     // RadixConnectMobile should require a NetworkAntenna and a SecureStorage from the Wallet.
-//     // The internal components, such as RadixConnectRelayService will be created by the RadixConnectMobile.
-//     #[uniffi::constructor]
-//     pub fn new(
-//         _network_antenna: Arc<dyn NetworkAntenna>,
-//         _secure_storage: Arc<dyn SecureStorage>,
-//     ) -> Self {
-//         todo!()
-//     }
-
-//     #[uniffi::method]
-//     pub fn handle_linking_request(
-//         &self,
-//         _request: RadixConnectMobileLinkRequest,
-//     ) -> Result<Url> {
-//         todo!()
-//     }
-
-//     #[uniffi::method]
-//     pub fn handle_dapp_interaction_request(
-//         &self,
-//         _request: RadixConnectMobileDappRequest,
-//     ) -> Result<DappToWalletInteraction> {
-//         todo!()
-//     }
-
-//     #[uniffi::method]
-//     pub fn send_dapp_interaction_response(
-//         &self,
-//         _response: WalletToDappInteractionResponse,
-//     ) -> Result<Url> {
-//         todo!()
-//     }
-// }
+#[uniffi::export]
+impl RadixConnectMobile {
+    #[uniffi::constructor]
+    pub fn new(
+        network_antenna: Arc<dyn NetworkAntenna>,
+        session_storage: Arc<dyn SessionStorage>,
+    ) -> Self {
+        Self::init(
+            Arc::new(RelayService::new_with_network_antenna(
+                network_antenna.clone(),
+            )),
+            session_storage,
+        )
+    }
+}
 
 #[uniffi::export]
-pub fn new_mobile_connect_request(
-    url: String,
-) -> Result<RadixConnectMobileConnectRequest> {
-    RadixConnectMobileConnectRequest::from_str(url.as_str())
+impl RadixConnectMobile {
+    #[uniffi::method]
+    pub async fn handle_deep_link(
+        &self,
+        url: String,
+    ) -> Result<RadixConnectMobileSessionRequest> {
+        // Try to parse the deep link ur to RadixConnectMobileRequest
+        let request = parse_mobile_connect_request(url)?;
+
+        // TODO: Move in parse_mobile_connect_request
+        request.verify_request_signature()?;
```

Comment:
relevant: `// TODO: Move in parse_mobile_connect_request` ?

Reply:
moved